lib/mathlib.cpp

```diff
@@ -548,6 +548,28 @@ MathLib::bigint MathLib::toLongNumber(const std::string & str)
     return ret;
 }
 
+// Assuming a limited support of built-in hexadecimal floats (see C99, C++17) that is a fall-back implementation.
+// For now it's not optimized WRT performance at all...
+static double FloatHexToDoubleNumber(const std::string& str)
+{
+    const std::size_t p = str.find('p');
+    const std::string number = str.substr(2, str.find('p') - 2);
+    const std::size_t decimal = number.find('.');
+    const std::string whole = number.substr(0, decimal);
+
+    double factor1 = std::stoi(whole, nullptr, 16);
+    if (std::string::npos != decimal) {
+        const std::string fraction = number.substr(decimal + 1);
+        const int b = std::stoi(fraction, nullptr, 16);
```

Comment:
It's probably best to not use `std::stoi` yet. It is not available in Slackware 14.0